server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java

```diff
@@ -154,13 +153,12 @@ protected Aggregator createInternal(SearchContext searchContext, Aggregator pare
         private ObjectArray<DuplicateByteSequenceSpotter> dupSequenceSpotters;
 
         SignificantTextCollectorSource(
-            SourceLookup sourceLookup,
             BigArrays bigArrays,
             MappedFieldType fieldType,
             String[] sourceFieldNames,
             boolean filterDuplicateText
         ) {
-            this.sourceLookup = sourceLookup;
+            this.sourceLookup = new SourceLookup();
```

Comment:
Since this one happens at query time maybe we can keep using it? I think what you have here would cause two significant text aggs to load source twice.

Reply:
Oops, this is a great point. I think I took it the refactor too far. I plan to restore the shared `SourceLookup` on `SearchLookup` so it can be used to share cached _source in the query phase.